Last stuff

TornaxO7 opened the pull request:
- Making the attributes of mboxes independent. We can store them now as well!
- Added more docs
- Added type-safety for flags
- Expanded flags a bit
- Added more tests
- Added a short summary of the file-structure in the beginning of the doc.

Please take a careful look into the `mbox/model.rs` file @soywod ! This is the last PR as far as I can tell. If I find something else, then I can make a PR on your repository after its merge ;)

TornaxO7 committed:
Last stuff

- Making the attributes of mboxes independent. We can store them now as well!
- Added more docs
- Added type-safety for flags
- Expanded flags a bit
- Added more tests
- Added a short summary of the file-structure in the beginning of the doc.

TornaxO7 committed:
Help command fix

Fixing help command description.

TornaxO7 committed:
Small doc change

TornaxO7 committed:
Doc fix

Fixing the link to the mbox delimiter.

TornaxO7 committed:
Fixing typo

TornaxO7 committed:
Doc fix

TornaxO7 committed:
Added docs for Output struct

soywod commented on src/imap/model.rs:
```diff
@@ -50,14 +70,41 @@ impl<'a> ImapConnector<'a> {
 
+    /// Closes the connection.
+    ///
+    /// **ALWAYS CALL THIS IF YOU DON'T NEED THE CONNECTION ANYMORE!**
```
Sounds like if you don't close the connection, a IMAP fanatic will come to your house and kill all your family :rofl: maybe the bold + capitalize is a bit too much.

TornaxO7 replied:
haha, you're right :sweat_smile:

TornaxO7 replied:
haha, you're right :sweat_smile:

soywod commented on src/imap/model.rs:
````diff
@@ -69,7 +116,30 @@ impl<'a> ImapConnector<'a> {
+    /// }
+    /// ```
+    pub fn add_flags(&mut self, mbox: &str, uid_seq: &str, flags: Flags) -> Result<()> {
+
````
Why adding a new line here? It does not really fit the global style of the code base.

TornaxO7 replied:
Should be fixed now.

soywod commented on src/imap/model.rs:
```diff
@@ -81,7 +151,12 @@ impl<'a> ImapConnector<'a> {
+    /// Remove the flags to the message by the given information. Take a look on the example above.
+    /// It's pretty similar.
+    pub fn remove_flags(&mut self, mbox: &str, uid_seq: &str, flags: Flags) -> Result<()> {
+
```
Why adding a new line here? It does not really fit the global style of the code base.

soywod commented on src/imap/model.rs:
````diff
@@ -50,14 +70,41 @@ impl<'a> ImapConnector<'a> {
+    /// }
+    /// ```
+    pub fn set_flags(&mut self, mbox: &str, uid_seq: &str, flags: Flags) -> Result<()> {
+
````
Why adding a new line here? It does not really fit the global style of the code base.

soywod commented on src/imap/model.rs:
```diff
@@ -15,13 +16,32 @@ error_chain! {
 }
 
 impl<'a> ImapConnector<'a> {
+
```
Why adding a new line here? It does not really fit the global style of the code base.

soywod commented on src/flag/model.rs:
```diff
@@ -124,3 +204,45 @@ fn convert_to_static<'func>(flag: &'func Flag) -> Result<Flag<'static>, ()> {
+
+#[cfg(test)]
+mod tests {
+
```
Why adding a new line here? It does not really fit the global style of the code base.

soywod commented on src/flag/model.rs:
```diff
@@ -73,6 +71,94 @@ impl ToString for Flags {
 
+impl ToString for Flags {
+    fn to_string(&self) -> String {
+
```
Why adding a new line here? It does not really fit the global style of the code base.

TornaxO7 committed:
Fixing tests

Fixing a little test issue

soywod commented on src/mbox/model.rs:
```diff
@@ -33,58 +34,71 @@ impl<'a> ser::Serialize for SerializableAttribute<'a> {
+            attributes.push_str(", ");
         }
+
+        // remove the trailing whitespace with the comma
```
What about [`.trim_end_matches()`](https://doc.rust-lang.org/std/string/struct.String.html#method.trim_end_matches)?

TornaxO7 replied:
well it's just a single whitespace, so I thought that pop might suit here. The compiler knows that he needs to remove only one character instead of looking after more whitespaces. At least I think that the compiler will think like this.

soywod replied:
But what if _one time_ there is no white space? Or if we have an empty `String`? The program will panic, where the `trim_end_matches` will not. I know it's a bit extreme, but the less panicking code we have the better we will be (in my opinion).

TornaxO7 replied:
Hm... ok, hold o!, I'll fix that quickly in a quick PR. (sorry, missread this part... ._.)

TornaxO7 committed:
Formatting changes + doc change

- Removed bold + capital words for logout-doc
- Run format on each *.rs file

TornaxO7 commented:
@soywod I threw `cargo fmt` on the project. second review pls

TornaxO7 commented:
Ok, I found the issue for the tests! The account has the smtp server etc. set to `None`. Hm... I think a suitable `test` trait as before might be suitable to avoid duplicated code. (since `imap_test` and `account` itself need a real-case `Account` struct).

TornaxO7 committed:
Fixing tests

- Testing the return value of the flags struct as a string doesn't really work
  since it's a HashSet => Converted it into a Vec (in the test) to set the order
  as well.
- Fixed imap test by reverting the changes in the test.

TornaxO7 committed:
Error handling

Changed error output when creating an Imap-Connection. Should help debugging :)

TornaxO7 commented:
@soywod tests are working now! You can review it again! :+1:

soywod reviewed (approved):
Looks good to me!
